### ISSUE TITLE
GroovyConfigWriter cannot be compiled with Java7

### ISSUE THREAD

**Issue by andersaaberg** [virtualdogbert/GroovyConfigWriter#3]
Grails3 supports Java7 (see http://docs.grails.org/latest/guide/gettingStarted.html) which is a requirement from my customer and Grails 3 uses GroovyConfigWriter, but the GroovyConfigWriter uses a Java8 specific feature which yields the following error:
```
| Error Failed to compile yml-to-groovy-config.groovy: com/virtualdogbert/GroovyConfigWriter : Unsupported major.minor version 52.0
java.lang.UnsupportedClassVersionError: com/virtualdogbert/GroovyConfigWriter : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:412)
	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:677)
	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:545)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:185)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:170)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:126)
	at org.codehaus.groovy.control.ResolveVisitor.resolveToOuter(ResolveVisitor.java:676)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:308)
	at org.codehaus.groovy.control.ResolveVisitor.visitClass(ResolveVisitor.java:1218)
	at org.codehaus.groovy.control.ResolveVisitor.startResolving(ResolveVisitor.java:176)
	at org.codehaus.groovy.control.CompilationUnit$11.call(CompilationUnit.java:651)
	at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:931)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:593)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:542)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:298)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:268)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:254)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:250)
	at org.grails.cli.profile.commands.factory.GroovyScriptCommandFactory.readCommandFile(GroovyScriptCommandFactory.groovy:48)
	at org.grails.cli.profile.commands.factory.GroovyScriptCommandFactory.readCommandFile(GroovyScriptCommandFactory.groovy)
	at org.grails.cli.profile.commands.factory.ResourceResolvingCommandFactory.findCommands(ResourceResolvingCommandFactory.groovy:44)
	at org.grails.cli.profile.commands.CommandRegistry.findCommands(CommandRegistry.groovy:86)
	at org.grails.cli.profile.commands.CommandRegistry.findCommands(CommandRegistry.groovy)
	at org.grails.cli.profile.AbstractProfile.getCommands(AbstractProfile.groovy:438)
	at org.grails.cli.profile.repository.AbstractJarProfileRepository$JarProfile.getCommands(AbstractJarProfileRepository.groovy:130)
	at org.grails.cli.profile.AbstractProfile.handleCommand(AbstractProfile.groovy:468)
	at org.grails.cli.GrailsCli.handleCommand(GrailsCli.groovy:378)
	at org.grails.cli.GrailsCli.handleCommand(GrailsCli.groovy:351)
	at org.grails.cli.GrailsCli.execute(GrailsCli.groovy:272)
	at org.grails.cli.GrailsCli.main(GrailsCli.groovy:159)
	at org.grails.cli.GrailsCli$main.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at grails.init.RunCommand.main(RunCommand.groovy:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at grails.init.Start.main(Start.java:127)
```

The problematic piece of code is the following from GroovyConfigWriter.groovy:
`if (!SourceVersion.RELEASE_8.isName(key) && !(quoteValues.contains(key))) {`

I suggest that RELEASE_8 is replaced by RELEASE_7 such that GroovyConfigWriter will be compatible with Grails 3 for us poor Java 7 dinosaurs. I will submit a pull request ASAP.

**Comment by virtualdogbert** [comment]
I accepted your change, and I bump and release a new version this weekend. Although do I have to set the source compatibility in the gradle.build, because I will be building with java 8?

You should tell your client that Java 7 end of life was April 2015, so no rush in upgrading :)
When I first started using Grails 3 I found it easier to upgrade to Java 8 than trying to get it running on Java 7, but I imagine clients can be stubborn.

**Comment by andersaaberg** [comment]
DOH yes of course - sourceCompatibility should also be set to 1.7 to avoid that other Java8 specific features are introduced later - I have made a new pull request with that: https://github.com/virtualdogbert/GroovyConfigWriter/pull/5

Many thanks for the quick reply - I will continue to persuade my customer to upgrade to Java8, but right now my main objective is to get Grails 3 at least :)

**Comment by virtualdogbert** [comment]
No prob, and yes Grails 3 is a nice step up.

**Comment by virtualdogbert** [comment]
So made the changes and released version 0.2. I also made a pull request to the external-config plugin.

**Comment by andersaaberg** [comment]
Many thanks for sending the fix upstream as well :+1:
